### ISSUE TITLE
route helper ignores query option

### ISSUE THREAD

**Issue by asiraky** [adonisjs/adonis-framework#1086]
## Package version
4.1.0

## Node.js and npm version
11.12.0 and 6.7.0

## Sample Code (to reproduce the issue)
```
Route.get('posts/:id', closure).as('posts.show');
route('posts.show', { id: 1 }, {
  query: { foo: 'bar' }
});
```
the route produced does not include the query as per the docs: https://adonisjs.com/docs/4.1/routing#_named_route

expected: 
```/posts/1?foo=bar```

received: 
```/posts/1```

## BONUS (a sample repo to reproduce the issue)
https://github.com/asiraky/adonis-route-bug

**Comment by asiraky** [comment]
Debugging the adonis source suggests the Route\Manager `url` method is missing sonething like this:

```javascript
let compiledRoute = pathToRegexp.compile(route._route)(data || {})

if (options.query) {
  compiledRoute = `${compiledRoute}?${Object.keys(options.query).map(k => `${k}=${encodeURIComponent(options.query[k])}`)}`
}
```

Hopefully this makes it easier for you.

**Comment by PurwadiPw** [comment]
Same issue here

**Comment by PurwadiPw** [comment]
> Debugging the adonis source suggests the Route\Manager `url` method is missing sonething like this:
> 
> ```js
> let compiledRoute = pathToRegexp.compile(route._route)(data || {})
> 
> if (options.query) {
>   compiledRoute = `${compiledRoute}?${Object.keys(options.query).map(k => `${k}=${encodeURIComponent(options.query[k])}`)}`
> }
> ```
> 
> Hopefully this makes it easier for you.

@asiraky

I get an error
```
"route" view global error: Assignment to constant variable.
```

**Comment by thetutlage** [comment]
Issue doesn't adhere to the guidelines of creating a proper issue.

1. Make sure that you are only reporting bugs or submitting feature requests on Github.
2. In case of feature requests, please be descriptive and explain how this feature helps a broader audience and not just you.

If you are not submitting a bug and neither a feature request, then please use the [forum](https://forum.adonisjs.com/) or [discord server](https://discord.gg/vDcEjq6?)


**In any case, issues which are not descriptive enough will be closed. Please respect our time.**

**Comment by asiraky** [comment]
Why are you closing this? This is a documented feature that does not work as documented. The issue was submitted using your template.

**Comment by thetutlage** [comment]
Ahh sorry, that’s been closed by mistake, since I was cleaning up old issues.

Mind sharing a PR for same?

**Comment by asiraky** [comment]
I have switched back to Laravel, sorry mate. I simply found too many issues like this one, with very little support from contributors. So after developing 25% of my app for around 3 months in Adonis, I have very little confidence in continuing with Adonis. Fortunately it only took around 1 week to convert everything to Laravel.

Much appreciated for trying to replicate what Laravel has in Javascript. Good effort but just not mature enough yet.
